rust/arrow/src/array/builder.rs

```diff
@@ -500,6 +500,26 @@ impl<T: ArrowPrimitiveType> PrimitiveBuilder<T> {
         Ok(())
     }
 
+    /// Appends values from a slice of type `T` and a validity byte slice
+    pub fn append_values(&mut self, values: &[T::Native], is_valid: &[u8]) -> Result<()> {
+        let ceil = bit_util::round_upto_power_of_2(values.len(), 8);
+        assert_eq!(
+            ceil / 8,
+            is_valid.len(),
+            "value slice not aligned with validity slice"
+        );
+        let mut bools = Vec::with_capacity(values.len() / 8 + 1);
+        is_valid.iter().for_each(|v| {
+            let bin = format!("{:b}", v);
+            // TODO: reversing to correct the endianness, find a better solution
+            bin.as_str().chars().rev().for_each(|c| {
+                bools.push(c == '1');
```

Comment:
Similar to the above comment, I don't know if there's perhaps a better way than formatting to binary and iterating through the binary characters.

Reply:
`{:b}` ignores leading zeros, which might not be want we want here. How about using bitmaks?

```rust
fn main() {
    // 0x38: 0011 1000
    // 0x1F: 0001 1111
    let bitmap = [0x38, 0x1F];
    bitmap.iter().enumerate().for_each(|(idx, byte)| {
        let mut bitmask = 0x01;
        println!("byte {}: {:b}", idx, byte);
        let offset = 0;
        for _ in offset..8 {
            println!("\t{}", (byte & bitmask) != 0);
            bitmask = bitmask << 1;
        }
    });
}
```
output:

```
byte 0: 111000
	false
	false
	false
	true
	true
	true
	false
	false
byte 1: 11111
	true
	true
	true
	true
	true
	false
	false
	false
```